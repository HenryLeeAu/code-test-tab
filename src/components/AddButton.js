import React, { Component } from 'react';
import { connect } from 'react-redux';
import { postNewMovie } from 'actions';
export class AddButton extends Component {
  handleClick = e => {
    this.props.postNewMovie();
  };
  render() {
    const { auth, searchStatus } = this.props;
    return auth && searchStatus.keyword.length <= 2 ? (
      <button onClick={this.handleClick} className="primary lg">
        +Add movie
      </button>
    ) : null;
  }
}
function mapStateToProps(state) {
  return {
    genreList: state.genreList,
    auth: state.auth,
    searchStatus: state.searchStatus,
  };
}

export default connect(
  mapStateToProps,
  { postNewMovie }
)(AddButton);
