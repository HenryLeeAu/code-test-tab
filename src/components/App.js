import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchRacingList } from 'actions';
class App extends Component {
  componentDidMount() {
    this.props.fetchRacingList();
  }
  render() {
    console.log(this.props.pageStatus.sorted_data);
    const newArr = this.props.pageStatus.sorted_data.filter(item => {
      return item.meeting.raceType === this.props.pageStatus.filter;
    });
    console.log(newArr);
    return <div className="App">home</div>;
  }
}

function mapStateToProps(state) {
  return {
    pageStatus: state.pageStatus,
  };
}

export default connect(
  mapStateToProps,
  { fetchRacingList }
)(App);
