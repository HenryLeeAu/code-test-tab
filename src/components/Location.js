import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updateLocation } from 'actions';
export class Location extends Component {
  componentDidMount() {}
  handleClick(location) {
    this.props.updateLocation({ location });
  }
  render() {
    const arr = ['NSW', 'VIC'];
    return (
      <ul className="tab location">
        {arr.map(location => {
          return (
            <li key={location}>
              <button
                className={
                  this.props.pageStatus.location === location ? 'clicked' : ''
                }
                onClick={e => this.handleClick(location)}
              >
                {location}
              </button>
            </li>
          );
        })}
      </ul>
    );
  }
}

function mapStateToProps(state) {
  return {
    pageStatus: state.pageStatus,
  };
}

export default connect(
  mapStateToProps,
  { updateLocation }
)(Location);
