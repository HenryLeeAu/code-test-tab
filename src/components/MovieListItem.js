import React from 'react';

export default ({
  auth,
  currentGenre,
  onClick,
  poster,
  title,
  genres,
  descripition,
}) => {
  return (
    <React.Fragment>
      <div className="postWrapper">
        {auth && currentGenre === null && (
          <button onClick={onClick} className="btnDelete waring sm">
            Delete
          </button>
        )}
        <img src={poster} alt={title} />
        {currentGenre === null && (
          <div className="tag">
            {genres.map(genre => (
              <span key={genre}>{genre}</span>
            ))}
          </div>
        )}
      </div>
      <h4>{title}</h4>
      <p>{descripition}</p>
    </React.Fragment>
  );
};
