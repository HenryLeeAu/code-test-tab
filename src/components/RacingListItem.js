import React from 'react';
import timeToGo from 'tools/timeToGo'
export default (props) =>{
  const {raceStartTime,raceName,raceNumber,meeting}=props.raceItem;
  const {meetingDate,meetingName,location} = meeting;

  return (
    <li>
      <button>
        <div className="raceNumber"><strong>R{raceNumber}</strong></div>
        <div><h2 className="raceName">{raceName}</h2></div>
        <div>{timeToGo(props.currentTime,raceStartTime)}</div>
        <div className="meetingName">{meetingName}({location})</div>
        <div className="meetingDate">{meetingDate}</div>
      </button>
    </li>
  )
}