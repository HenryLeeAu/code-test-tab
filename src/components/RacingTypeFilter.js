import React, { Component } from 'react';
import { connect } from 'react-redux';
import { updateRacingType } from 'actions';
export class RaceTypeFilter extends Component {
  componentDidMount() {}
  handleClick(racingType) {
    this.props.updateRacingType({ racingType });
  }
  render() {
    const arr = [
      { filterWord: 'R', text: 'Thoroughbred' },
      { filterWord: 'G', text: 'Greyhounds' },
      { filterWord: 'H', text: 'Harness' },
    ];

    return (
      <ul className="tab type">
        {arr.map(tab => {
          const { filterWord, text } = tab;
          return (
            <li key={text}>
              <button
                onClick={e => this.handleClick(filterWord)}
                className={
                  this.props.pageStatus.filter === filterWord ? 'clicked' : ''
                }
              >
                {text}
              </button>
            </li>
          );
        })}
      </ul>
    );
  }
}

function mapStateToProps(state) {
  return {
    pageStatus: state.pageStatus,
  };
}

export default connect(
  mapStateToProps,
  { updateRacingType }
)(RaceTypeFilter);
