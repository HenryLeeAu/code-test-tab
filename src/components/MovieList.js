import React, { Component } from 'react';
import MovieListItem from 'components/MovieListItem';
import { connect } from 'react-redux';
import { fetchMovieList, deleteMovie } from 'actions';

export class MovieList extends Component {
  handleDelete(id) {
    this.props.deleteMovie(id);
  }
  componentDidMount() {
    this.props.fetchMovieList();
  }
  renderList() {
    return this.props.searchStatus.currentMovieList.map((movie, index) => {
      return (
        <li key={`${movie.id}${movie.title}`}>
          <MovieListItem
            onClick={e => this.handleDelete(movie.id)}
            title={movie.title}
            poster={movie.poster}
            genres={movie.genres}
            descripition={movie.descripition}
            auth={this.props.auth}
            currentGenre={this.props.searchStatus.currentGenre}
            clicked={movie.id === this.props.searchStatus.currentId}
          />
        </li>
      );
    });
  }
  render() {
    return this.props.searchStatus.currentMovieList.length > 0 ? (
      <ul className="movieList">{this.renderList()}</ul>
    ) : (
      <div className="noresult">No result</div>
    );
  }
}
function mapStateToProps(state) {
  return {
    searchStatus: state.searchStatus,
    auth: state.auth,
  };
}

export default connect(
  mapStateToProps,
  { fetchMovieList, deleteMovie }
)(MovieList);
