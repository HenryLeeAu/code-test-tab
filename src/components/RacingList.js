import React, { Component } from 'react';
import RacingListItem from 'components/RacingListItem';
import { connect } from 'react-redux';
import { fetchRacingList } from 'actions';
export class RacingList extends Component {
  state = {
    currentTime: new Date().getTime(),
  };
 
  componentDidMount() {
    this.props.fetchRacingList({location:'NSW'});
    this.interval = setInterval(() => {
      this.setState({
        currentTime: new Date().getTime(),
      });
    }, 60000);
  }

  componentWillUnmount(){
    clearInterval(this.interval);
    
  }
  render() {
    const filteredList = this.props.pageStatus.sorted_data.filter(item => {
      return item.meeting.raceType === this.props.pageStatus.filter;
    });
    if (filteredList.length) {
      return (
        <ul className="list">
          {filteredList.map((raceItem, index) => {
            return (
              <RacingListItem
                raceItem={raceItem}
                key={index + raceItem.raceName}
                currentTime={this.state.currentTime}
              />
            );
          })}
        </ul>
      );
    } else {
      return <div id="noresult">Noresult</div>;
    }
  }
}
function mapStateToProps(state) {
  return {
    pageStatus: state.pageStatus,
  };
}

export default connect(
  mapStateToProps,
  { fetchRacingList }
)(RacingList);
