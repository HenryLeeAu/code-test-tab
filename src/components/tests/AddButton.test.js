import React from 'react';
import { AddButton } from 'components/AddButton';
import { shallow } from 'enzyme';

let wrappered;

describe('AddButton', () => {
  it('auth true and keyword < 2 ', () => {
    const props = {
      auth: true,
      searchStatus: {
        keyword: {
          keyword: 'ab',
          length: 2,
        },
      },
    };
    wrappered = shallow(<AddButton {...props} />);
    expect(wrappered).toMatchSnapshot();
    expect(wrappered.find('button').length).toBe(1);
  });
  it('auth false', () => {
    const props = {
      auth: false,
    };
    wrappered = shallow(<AddButton {...props} />);
    expect(wrappered).toMatchSnapshot();
    expect(wrappered.find('button').length).toBe(0);
  });
  it('auth true and keyword > 2 ', () => {
    const props = {
      auth: true,
      searchStatus: {
        keyword: {
          keyword: 'abc',
          length: 3,
        },
      },
    };
    wrappered = shallow(<AddButton {...props} />);
    expect(wrappered).toMatchSnapshot();
    expect(wrappered.find('button').length).toBe(0);
  });
});
