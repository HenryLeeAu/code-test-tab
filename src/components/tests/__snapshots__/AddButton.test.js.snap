// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton auth false 1`] = `""`;

exports[`AddButton auth true and keyword < 2  1`] = `
<button
  className="primary lg"
  onClick={[Function]}
>
  +Add movie
</button>
`;

exports[`AddButton auth true and keyword > 2  1`] = `""`;
